addons/sale_coupon/models/sale_order.py

```diff
@@ -80,19 +80,20 @@ def _get_reward_values_product(self, program):
 
         order_lines = (self.order_line - self._get_reward_lines()).filtered(lambda x: program._get_valid_products(x.product_id))
         max_product_qty = sum(order_lines.mapped('product_uom_qty')) or 1
+        best_qty = sum(self.order_line.filtered(lambda x: x.product_id == program.reward_product_id).mapped('product_uom_qty'))
```

Comment:
I think this is more meaningful:
```suggestion
        total_qty = sum(self.order_line.filtered(lambda x: x.product_id == program.reward_product_id).mapped('product_uom_qty'))
```